crates/ra_hir/src/ty/tests.rs

```diff
@@ -2417,6 +2417,31 @@ fn test() -> u64 {
     );
 }
 
+#[test]
+fn infer_macros_expanded() {
+    assert_snapshot_matches!(
+        infer(r#"
+macro_rules! vec {
+    ($($item:expr),*) => {
+            {
+                let mut v = Vec::new();
+                $(
+                    v.push($item);
+                )*
+                    v
+            }
+    };
+}
+fn main() {
+    let x = vec![1,2];
+}
+"#),
+        @r###"
+[229; 255) '{     ...,2]; }': ()
+[239; 240) 'x': {unknown}"###
```

Comment:
Let's use a test which gives something other than `{unknown}`? One way to do that would be to add `struct Vec` and the corresponding `impl` here (not the real vec, just a minimal thing to make code type check). If that fails due to something broken eslsewhere, we can try a more minimal test with a macro whihc does nothing `($e:expr) => { $e }`